### DISCUSSION TITLE
Regarding including Reservoirs in HYMAP

### DISCUSSION THREAD

**Discussion by PrateekSharma99** [NASA-LIS/LISF#1217] [Q&A]
I am trying to include a single reservoir data in to the HYMAP part of LIS. Following is the command setup that I have written. Could you suggest me as to why am I getting the following errors.

TEMPLATE model timestep:                      "1hr" 
HYMAP2 routing model time step:               "1hr"
HYMAP2 routing model output interval:         "1da"
HYMAP2 routing model restart interval:        "1mo"
HYMAP2 routing method:                            "local inertia" "
HYMAP2 routing model time step method:       "adaptive" 
HYMAP2 routing model adaptive time step alfa coefficient: 1.0
HYMAP2 run in ensemble mode:                 1
HYMAP2 routing model linear reservoir flag:  1
HYMAP2 routing model evaporation option:     none 
HYMAP2 routing model restart file:      *LOCATION*
HYMAP2 routing model start mode:             coldstart 

HYMAP2 floodplain dynamics: 0  
HYMAP2 reservoir operation option: 1 
HYMAP2 number of reservoirs: 1
HYMAP2 reservoir operation input time series size: 150
HYMAP2 reservoir operation input directory: *LOCATION*
HYMAP2 reservoir operation header filename: *LOCATION*
HYMAP2 reservoir operation input data type: "water level"
HYMAP2 enable 2-way coupling: 0
HYMAP2 routing model dwi flag: 0

Whenever I try to run this part I am facing  this error could anyone tell me why is it so

forrtl: error (65): floating invalid
Image              PC                Routine            Line        Source             
LIS                00000000013CFCD4  Unknown               Unknown  Unknown
libpthread-2.17.s  00002ADE5D65F630  Unknown               Unknown  Unknown
LIS                00000000004A7048  hymap2_resopmod_m          56  HYMAP2_resopMod.F90
LIS                00000000004A54DA  hymap2_model_core         330  HYMAP2_model_core.F90
LIS                00000000004A054E  hymap2_model_             229  HYMAP2_model.F90
LIS                00000000004C9466  hymap2_routing_ru         349  HYMAP2_routing_run.F90
LIS                00000000006838D9  lis_routingmod_mp         131  LIS_routingMod.F90
LIS                000000000123A583  retrospective_run         206  retrospective_runMod.F90
LIS                0000000000CC6014  MAIN__                     68  lisdrv.F90
LIS                0000000000416632  Unknown               Unknown  Unknown
libc-2.17.so       00002ADE5DB90555  __libc_start_main     Unknown  Unknown
LIS                0000000000416529  Unknown               Unknown  Unknown

**Comment by agetiran** [comment]
@Daduji123,
you need to specify the directory and header files for the reservoir operation scheme. It's not clear what "LOCATION" means.

HYMAP2 reservoir operation input directory: LOCATION
HYMAP2 reservoir operation header filename: LOCATION

In the reservoir operation input directory, you'll need a text file for each reservoir with two columns: date (YYYYMMDD), and reservoir water level observations. The reservoir operation header filename is a text file containing three columns: reservoir name, i.e., same name of text files used for each reservoir (but without extension), x, y. x and y are the horizontal and vertical locations of the grid points in the domain.

**Reply by PrateekSharma99** [reply]
Good Afternoon @agetiran The location is specified already as follows: 
![image](https://user-images.githubusercontent.com/89891132/216530776-ae34c95a-2d3d-4485-a486-987c5cbec2aa.png)
I was able to solve the earlier error by changing some input parameters. The model fails at a point where the outputs are written after a single cycle run for the routing part whenever reservoirs are included but works fine without the reservoirs. This type of error occurs: 
![image](https://user-images.githubusercontent.com/89891132/216531177-d1c091e4-081c-4edd-bc7d-13b34ecfb4a4.png)
I was able to point out that the error occurs in the **sub-routine LIS_writeRoutingModelOutput of LIS_historyMod.F90   LINE:(614-750 ) specific line number: 733,738**
I have attached the MODEL_OUTPUT_LIST_FAT and the .config file that I am using. Could you please provide any inputs on this.
[Prateek.zip](https://github.com/NASA-LIS/LISF/files/10576326/Prateek.zip)
